### PR TITLE
Add globalnet flag for lighthouse

### PR DESCRIPTION
**Pull request by vthapar**
Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

### CODE REVIEW

File: pkg/apis/submariner/v1alpha1/servicediscovery_types.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ type ServiceDiscoverySpec struct {
 	Debug                    bool   `json:"debug"`
 	ClusterID                string `json:"clusterID"`
 	Namespace                string `json:"namespace"`
+	GlobalnetEnabled         bool   `json:"globalnetEnabled,omitempty"`
```
Comment (mangelajo) [inline pkg/apis/submariner/v1alpha1/servicediscovery_types.go:38]: I think omitempty needs to be a pointer to bool.

**Comment by mangelajo** [comment]
> I think it would be fine, but we need to check what happens with the CRD API version, do we need a bump for this?, and if so.. we would need the translator webhook for CRDs.
> 
> I suspect that being additive and optional field makes that change fine, specially in Alpha. check my comment about "omitempty"

Also, I wonder, if we start relying on this flag, then when the operator is updated, for example 0.4.0 to 0.5.0, we'd need to reprocess the "Submariner CR" into the "ServiceDiscovery CR" updating that field.

Can we try something like this manually? 
1) install with subctl release 0.4.1 ...
2) push this operator image somewhere, and compile the new subctl
3) run subctl join again .. ? 

I suspect it would work.

File: pkg/apis/submariner/v1alpha1/servicediscovery_types.go (hunk 1 above)
Comment (vthapar) [reply pkg/apis/submariner/v1alpha1/servicediscovery_types.go:38]: > I think omitempty needs to be a pointer to bool.

We're not using pointer for other bool `omitempty`

**Comment by vthapar** [comment]
> > I think it would be fine, but we need to check what happens with the CRD API version, do we need a bump for this?, and if so.. we would need the translator webhook for CRDs.
> > I suspect that being additive and optional field makes that change fine, specially in Alpha. check my comment about "omitempty"
> 
> Also, I wonder, if we start relying on this flag, then when the operator is updated, for example 0.4.0 to 0.5.0, we'd need to reprocess the "Submariner CR" into the "ServiceDiscovery CR" updating that field.
> 
> Can we try something like this manually?
> 
> 1. install with subctl release 0.4.1 ...
> 2. push this operator image somewhere, and compile the new subctl
> 3. run subctl join again .. but specifying the image that you pushed on 2.
> 
> I suspect it would work.

It didn't work coz we don't really have reconcile logic yet. But this is something that can be handled easily when we do have an upgrade story in place. As an extra precaution, I'll use configMap in Lighthouse code if the flag is missing/nil and then we can remove that check once we have upgrade/migration sorted out.

**Comment by mangelajo** [comment]
> > > I think it would be fine, but we need to check what happens with the CRD API version, do we need a bump for this?, and if so.. we would need the translator webhook for CRDs.
> > > I suspect that being additive and optional field makes that change fine, specially in Alpha. check my comment about "omitempty"
> > 
> > 
> > Also, I wonder, if we start relying on this flag, then when the operator is updated, for example 0.4.0 to 0.5.0, we'd need to reprocess the "Submariner CR" into the "ServiceDiscovery CR" updating that field.
> > Can we try something like this manually?
> > 
> > 1. install with subctl release 0.4.1 ...
> > 2. push this operator image somewhere, and compile the new subctl
> > 3. run subctl join again .. but specifying the image that you pushed on 2.
> > 
> > I suspect it would work.
> 
> It didn't work coz we don't really have reconcile logic yet. But this is something that can be handled easily when we do have an upgrade story in place. As an extra precaution, I'll use configMap in Lighthouse code if the flag is missing/nil and then we can remove that check once we have upgrade/migration sorted out.

Ack I added this for a follow up: https://github.com/submariner-io/submariner-operator/issues/510

**Pull request merged**